osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/periodicmeterreads/GetPeriodicMeterReadsCommandExecutor.java

```diff
@@ -90,160 +71,207 @@ public PeriodicMeterReadsRequestDto fromBundleRequestInput(final ActionRequestDt
+        allAttributeAddresses.add(profileBufferAddress.getAttributeAddress());
+        allAttributeAddresses.addAll(scalerUnitAddresses);
+
+        for (final AttributeAddress address : allAttributeAddresses) {
+
+            conn.getDlmsMessageListener()
+                    .setDescription(String.format(FORMAT_DESCRIPTION, queryPeriodType, from, to,
+                            JdlmsObjectToStringUtil.describeAttributes(address)));
+
+            getResultList.addAll(this.dlmsHelper.getAndCheck(conn, device,
+                    "retrieve periodic meter reads for " + queryPeriodType, address));
         }
 
+        LOGGER.info("Received getResult: {} ", getResultList);
+
         final DataObject resultData = this.dlmsHelper.readDataObject(getResultList.get(0), PERIODIC_E_METER_READS);
         final List<DataObject> bufferedObjectsList = resultData.getValue();
+
         final List<PeriodicMeterReadsResponseItemDto> periodicMeterReads = new ArrayList<>();
         for (final DataObject bufferedObject : bufferedObjectsList) {
+            final List<DataObject> bufferedObjectValue = bufferedObject.getValue();
+
             try {
-                periodicMeterReads.add(this
-                        .processNextPeriodicMeterReads(periodType, from, to, bufferedObject.getValue(), getResultList));
+                periodicMeterReads.add(
+                        this.convertToResponseItem(periodicMeterReadsQuery, bufferedObjectValue,
+                                getResultList, profileBufferAddress, scalerUnitAddresses, periodicMeterReads, intervalTime));
             } catch (final BufferedDateTimeValidationException e) {
                 LOGGER.warn(e.getMessage(), e);
             }
         }
 
-        return new PeriodicMeterReadsResponseDto(periodType, periodicMeterReads);
+        return new PeriodicMeterReadsResponseDto(queryPeriodType, periodicMeterReads);
     }
 
-    private PeriodicMeterReadsResponseItemDto processNextPeriodicMeterReads(final PeriodTypeDto periodType,
-            final DateTime from, final DateTime to, final List<DataObject> bufferedObjects,
-            final List<GetResult> results) throws ProtocolAdapterException, BufferedDateTimeValidationException {
-
-        final DateTime bufferedDateTime = this.getBufferedDateTime(periodType, from, to, bufferedObjects);
-
-        switch (periodType) {
-        case INTERVAL:
-            return this.getNextPeriodicMeterReadsForInterval(bufferedObjects, bufferedDateTime, results);
-        case DAILY:
-            return this.getNextPeriodicMeterReadsForDaily(bufferedObjects, bufferedDateTime, results);
-        case MONTHLY:
-            return this.getNextPeriodicMeterReadsForMonthly(bufferedObjects, bufferedDateTime, results);
-        default:
-            throw new AssertionError("Unknown PeriodType: " + periodType);
+    private PeriodicMeterReadsResponseItemDto convertToResponseItem(
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Method has 7 parameters, which is greater than 6 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS00107)